X-Gentoo-Bugzilla: https://bugs.gentoo.org/show_bug.cgi?id=443666
X-Git-Url: http://lxde.git.sourceforge.net/git/gitweb.cgi?p=lxde%2Flxdm;a=blobdiff_plain;f=configure.ac;h=9e21a22d893b45f1564e23b12c00a0bf546d95e8;hp=3c540df5cc76aa8c3ef394bae4f0d0cf33bff993;hb=371b018474217bdff1d5108b384fbf0f45aa873d;hpb=f625fd7b1f90caac260422a6c13e0cb12ccf1f96

Further changes: Move the AC_CHECK_LIB([ck-connector]) inside the 'if' block
to avoid checking and enabling ck-connector even if --disable-consolekit was
passed.

Signed-off-by: Markos Chandras <hwoarang@gentoo.org>

Index: lxdm-0.4.1/configure.ac
===================================================================
--- lxdm-0.4.1.orig/configure.ac
+++ lxdm-0.4.1/configure.ac
@@ -28,8 +28,6 @@ if test "x$USE_PAM" != "xno" ; then
 		 fi])
 fi
 
-AC_CHECK_LIB([ck-connector],[ck_connector_open_session])
-
 # Checks for header files.
 AC_PATH_X
 AC_CHECK_HEADERS([shadow.h stdlib.h string.h unistd.h utmpx.h])
@@ -89,11 +87,22 @@ PKG_CHECK_MODULES(GTK, [$pkg_modules])
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(CONSOLEKIT, "ck-connector",[
+
+AC_ARG_ENABLE(consolekit,
+	AC_HELP_STRING([--disable-consolekit],[disable ConsoleKit support]),
+	[enable_consolekit=$enableval],
+	[enable_consolekit=yes]
+)
+
+if test "x$enable_consolekit" = "xyes" ; then
+  AC_CHECK_LIB([ck-connector],[ck_connector_open_session])
+  PKG_CHECK_MODULES(CONSOLEKIT, "ck-connector",[
 	AC_SUBST(CONSOLEKIT_CFLAGS)
 	AC_SUBST(CONSOLEKIT_LIBS)],[
 	echo "ConsoleKit devel package not found"
 	])
+fi
+
 
 AC_ARG_ENABLE(password,
 AC_HELP_STRING([--enable-password],[enable to load autologin password store at config file]),
